app/src/main/java/com/quipper/android/apollofrontpage/repository/impl/PostsRepository.kt

```diff
@@ -0,0 +1,53 @@
+package com.quipper.android.apollofrontpage.repository.impl
+
+import android.annotation.SuppressLint
+import androidx.lifecycle.MutableLiveData
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.rx2.rxMutate
+import com.apollographql.apollo.rx2.rxQuery
+import com.quipper.android.apollofrontpage.AllPostsQuery
+import com.quipper.android.apollofrontpage.UpvotePostMutation
+import com.quipper.android.apollofrontpage.fragment.PostDetails
+import com.quipper.android.apollofrontpage.model.PostsResult
+import com.quipper.android.apollofrontpage.repository.PostsRepository
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers.io
+
+class PostsRepository(
+    private val apolloClient: ApolloClient
+) : PostsRepository {
+    private var postsData = MutableLiveData<List<PostDetails>>()
```

Comment:
[IMO] We'd better use RxJava to handle results. Because it's not the same timing returning LiveData and API result. I find it difficult to follow the data-flow.

Reply:
I Agree. We should not have variables in the Repository for data access. Since Repositories are singleton, the variables become global ones and not an elegant solution to access data that changes over time.